### ISSUE TITLE
unexpected warnings: warning: failed to compute relocation: R_RISCV_SET_ULEB128, Invalid data was encountered while parsing the file

### ISSUE THREAD

**Issue by BaoshanPang** [llvm/llvm-project#101544]
```
$ cat t.c
int t(int x, int y){
  if(x > y){
    asm("add x0, x0, x0");
    return 1;
  }
  else
    return 0;
}

$ ./bin/clang -O2 -g -c t.c

$ ./bin/llvm-objdump -d -l -S t.o > t.obj
warning: failed to compute relocation: R_RISCV_SET_ULEB128, Invalid data was encountered while parsing the file
warning: failed to compute relocation: R_RISCV_SUB_ULEB128, Invalid data was encountered while parsing the file
warning: failed to compute relocation: R_RISCV_SET_ULEB128, Invalid data was encountered while parsing the file
warning: failed to compute relocation: R_RISCV_SUB_ULEB128, Invalid data was encountered while parsing the file
warning: failed to compute relocation: R_RISCV_SET_ULEB128, Invalid data was encountered while parsing the file
warning: failed to compute relocation: R_RISCV_SUB_ULEB128, Invalid data was encountered while parsing the file

```

**Comment by BaoshanPang** [comment]
$ ./bin/clang --version
clang version 20.0.0git (https://github.com/llvm/llvm-project.git 855703537e4d7eb1d7310ba40c3459e79e3243d2)
Target: riscv32-unknown-unknown-elf
Thread model: posix

**Comment by BaoshanPang** [comment]
$ ./bin/llvm-objdump --version
LLVM (http://llvm.org/):
  LLVM version 20.0.0git
  Optimized build.


  Registered Targets:
    riscv32 - 32-bit RISC-V
    riscv64 - 64-bit RISC-V

**Comment by topperc** [comment]
CC @MaskRay
